libminifi/src/processors/ExtractText.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * @file ExtractText.cpp
+ * ExtractText class implementation
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <iterator>
+#include <string>
+#include <memory>
+#include <set>
+
+#include "processors/ExtractText.h"
+#include "core/ProcessContext.h"
+#include "core/ProcessSession.h"
+#include "core/FlowFile.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+core::Property ExtractText::Attribute("Attribute", "Attribute to set from content (TEMPORARY)", "");
+core::Relationship ExtractText::Success("success", "success operational on the flow record");
+
+void ExtractText::initialize() {
+    //! Set the supported properties
+    std::set<core::Property> properties;
+    properties.insert(Attribute);
+    setSupportedProperties(properties);
+    //! Set the supported relationships
+    std::set<core::Relationship> relationships;
+    relationships.insert(Success);
+    setSupportedRelationships(relationships);
+}
+
+void ExtractText::onTrigger(core::ProcessContext *context, core::ProcessSession *session) {
+    std::shared_ptr<core::FlowFile> flowFile = session->get();
+
+    if (!flowFile) {
+        return;
+    }
+
+    ReadCallback cb(flowFile, context);
+    session->read(flowFile, &cb);
+    session->transfer(flowFile, Success);
+}
+
+int64_t ExtractText::ReadCallback::process(std::shared_ptr<io::BaseStream> stream) {
+    int64_t ret = 0;
+    uint64_t read_size = 0;
+
+    std::string attrKey;
+    _ctx->getProperty(Attribute.getName(), attrKey);
+    std::stringstream contentStream(std::stringstream::out | std::stringstream::in);
+    std::string contentStr;
+
+    while (read_size < _flowFile->getSize()) {
+        ret = stream->read(_buffer, _max_read);
+        if (ret < 0) {
+            return -1;
+        }
+
+        if (ret > 0) {
+            contentStream.write(reinterpret_cast<const char*>(_buffer), ret);
```

Comment:
I see that this appears to be a partial implementation of the JAVA class; however, one component that I think is necessary will be the limits on write. As it stands, the stringstream is unbounded based on the size of the input file. Don't you think we should bound or cap this with a configurable limit to avoid memory issues when users aren't always aware of the size of the content coming through?

Reply:
Sure. Should it be a constant somewhere, or a property?

Reply:
I think we should add that as a processor property. Something like "Size Limit" with a good description so that the user becomes conscious of the cost.